File: src/main/java/nuclearcoder/discordbot/command/misc/CmdPat.java
package nuclearcoder.discordbot.command.misc;

import nuclearcoder.discordbot.NuclearBot;
import nuclearcoder.discordbot.command.Command;
import sx.blah.discord.handle.obj.IChannel;
import sx.blah.discord.handle.obj.IMessage;
import sx.blah.discord.handle.obj.IUser;

import java.util.Arrays;

public class CmdPat implements Command {

    @Override public void execute(NuclearBot bot, IMessage message, String command, String[] args)
    {
        if (args.length >= 2)
        {
            final IChannel channel = message.getChannel();
            final IUser user = message.getAuthor();

            final String target = String.join(" ", Arrays.copyOfRange(args, 1, args.length)).trim();

            bot.sendMessage(channel, user.mention() + " pats " + target);
        }
    }

}

File: src/main/java/nukebot/command/misc/CmdPat.kt
package nukebot.command.misc

import nukebot.NuclearBot
import nukebot.command.Command
import nukebot.rSendMessage
import sx.blah.discord.handle.obj.IMessage
import java.util.*

class CmdPat : Command {

    override fun execute(bot: NuclearBot, message: IMessage, command: String, args: Array<String>) {
        if (args.size >= 2) {
            val channel = message.channel
            val user = message.author

            val target = Arrays.copyOfRange(args, 1, args.size).joinToString(" ").trim { it <= ' ' }

            channel.rSendMessage(user.mention() + " pats " + target)
        }
    }

}
